### PR TITLE
feat: Huobi provider to WS

### PR DESCRIPTION
**Pull request by RafilxTenfen**
## Description

- Modified Huobi provider to use [WebSocket](https://huobiapi.github.io/docs/spot/v1/en/#websocket-market-data) connection instead of `http.client`
- Modified unit tests for checking the map

ref: #509

----

### Author Checklist

*All items are required. Please add a note to the item if the item is not applicable and
please add links to any relevant follow up issues.*

I have...

- [ ] included the correct [type prefix](https://github.com/commitizen/conventional-commit-types/blob/v3.0.0/index.json) in the PR title
- [ ] added appropriate labels to the PR
- [ ] added `!` to the type prefix if API or client breaking change
- [ ] targeted the correct branch (see [PR Targeting](https://github.com/umee-network/umee/blob/main/CONTRIBUTING.md#pr-targeting))
- [ ] provided a link to the relevant issue or specification
- [ ] added a changelog entry to `CHANGELOG.md`
- [ ] included comments for [documenting Go code](https://blog.golang.org/godoc)
- [ ] updated the relevant documentation or specification
- [ ] reviewed "Files changed" and left comments if necessary
- [ ] confirmed all CI checks have passed

### Reviewers Checklist

*All items are required. Please add a note if the item is not applicable and please add
your handle next to the items reviewed if you only reviewed selected items.*

I have...

- [ ] confirmed the correct [type prefix](https://github.com/commitizen/conventional-commit-types/blob/v3.0.0/index.json) in the PR title
- [ ] confirmed `!` in the type prefix if API or client breaking change
- [ ] confirmed all author checklist items have been addressed
- [ ] reviewed state machine logic
- [ ] reviewed API design and naming
- [ ] reviewed documentation is accurate
- [ ] reviewed tests and test coverage
- [ ] manually tested (if applicable)

### CODE REVIEW

File: price-feeder/oracle/provider/huobi.go [hunk 1]
```diff
@@ -27,140 +31,214 @@ type (
...
 	}
 
-	rawPrice := tradeResp.Tick.Data[0].Price
-	price, err := sdk.NewDecFromStr(strconv.FormatFloat(rawPrice, 'f', -1, 64))
-	if err != nil {
-		return TickerPrice{}, fmt.Errorf("failed to parse Huobi price (%f) for %s", rawPrice, ticker)
+	p.setTickerPair(tickerResp)
+}
+
+// pong return a ping message sent by the provider as pong
```
Comment (adamewozniak) [inline price-feeder/oracle/provider/huobi.go:159]:
```suggestion
// pong return a heartbeat message when a "ping" is received
```

File: price-feeder/oracle/provider/huobi.go [hunk 2]
```diff
@@ -27,140 +31,214 @@ type (
...
+	return tickerPrices, nil
+}
 
-	bz, err := ioutil.ReadAll(resp.Body)
-	if err != nil {
-		return sdk.ZeroDec(), fmt.Errorf("failed to read Huobi response body: %w", err)
+func (p *HuobiProvider) getTickerPrice(cp types.CurrencyPair) (TickerPrice, error) {
+	ticker, ok := p.tickers[getChannelTicker(cp)]
+	if !ok {
+		return TickerPrice{}, fmt.Errorf("failed to get %s", cp.String())
```
Comment (adamewozniak) [inline price-feeder/oracle/provider/huobi.go:207]:
```suggestion
		return TickerPrice{}, fmt.Errorf("failed to get ticker price for %s", cp.String())
```

File: price-feeder/oracle/provider/huobi.go [hunk 3]
```diff
@@ -27,140 +31,214 @@ type (
...
+}
 
-	if !strings.EqualFold(marketResp.Status, "ok") {
-		return sdk.ZeroDec(), fmt.Errorf(
-			"received unexpected error from Huobi response: %s",
-			marketResp.ErrMsg,
-		)
+// uncompressGzip uncompress gzip compressed messages
+// All return data of websocket Market APIs are compressed
+// with GZIP so they need to be unzipped.
```
Comment (adamewozniak) [inline price-feeder/oracle/provider/huobi.go:215]:
```suggestion
// All data returned from the websocket Market APIs is compressed
// with GZIP, so it needs to be unzipped.
```

**Review by adamewozniak — COMMENTED** [review]
just a couple nits, plus : 
1. Please add a line to `CHANGELOG.md` 
2. Why do we not need reconnection logic like we have with binance & okx?

**Comment by RafilxTenfen** [comment]
> * Please add a line to `CHANGELOG.md`

My bad forgot it again :/

**Comment by adamewozniak** [comment]
> @adamewozniak
> 
> > 2. Why do we not need reconnection logic like we have with binance & okx?
> 
> * Huobi handles it differently, it maintains the connection alive without any time constraint, while we still respond to the heartbeat message [ping x pong](https://github.com/umee-network/umee/pull/569/files#diff-88966ac3f8eb0c4b434ceef0aac6226c51a21863af1602e40cc085ddaf14e0ecR166)
> 
> [huobi doc](https://huobiapi.github.io/docs/spot/v1/en/#introduction-10)

Per the huobi docs : 
> After the server sent two consecutive heartbeat messages without receiving at least one matching "pong" response from a client, then right before server sends the next "ping" heartbeat, the server will be disconnected with the client server.

So then, if for some reason the price feeder didn't send the pong response to huobi within that window, the user would have to restart the program to reconnect? In the case of an intermittent outage, or the price feeder is unable to send that response for whatever reason, we wouldn't want to require validators to restart the price feeder in order to reconnect huobi

**Comment by RafilxTenfen** [comment]
> So then, if for some reason the price feeder didn't send the pong response to huobi within that window, the user would have to restart the program to reconnect? In the case of an intermittent outage, or the price feeder is unable to send that response for whatever reason, we wouldn't want to require validators to restart the price feeder in order to reconnect huobi 

- ohh yeah, you are correct, thanks for clarifying 
- I will think of something to handle that 
- until there I will set the PR as draft

**Comment by RafilxTenfen** [comment]
@adamewozniak 

I did 2 things to try to reconnect, one of them is as usual, try to reconnect at each 2 min if the ticker is not reset by a ping message, the other one became a litter [weird](https://github.com/umee-network/umee/pull/569/files#diff-88966ac3f8eb0c4b434ceef0aac6226c51a21863af1602e40cc085ddaf14e0ecR114) 
- if a message can not be read it generates an error in this case we try to ping the provider to check the connection 
   - if the ping returns an error it means that the connection is closed so I try to reconnect there too

Any thoughts around that?

**Comment by adamewozniak** [comment]
> @adamewozniak
> 
> I did 2 things to try to reconnect, one of them is as usual, try to reconnect at each 2 min if the ticker is not reset by a ping message, the other one became a little [weird](https://github.com/umee-network/umee/pull/569/files#diff-88966ac3f8eb0c4b434ceef0aac6226c51a21863af1602e40cc085ddaf14e0ecR114)
> 
> * if a message can not be read it generates an error in this case we try to ping the provider to check the connection
>   
>   * if the ping returns an error it means that the connection is closed so I try to reconnect there too
> 
> Any thoughts around that?

I think it makes sense to handle the case this way - I'm assuming the err coming out of `p.wsClient.ReadMessage()` is rare / would only happen if the provider was down, yes?

File: price-feeder/oracle/provider/huobi.go [hunk 4]
```diff
@@ -27,140 +32,248 @@ type (
...
+		case <-time.After(defaultReadNewMessage):
+			// time after to avoid asking for prices too frequently
+			messageType, bz, err := p.wsClient.ReadMessage()
+			if err != nil {
+				// if some error occurs continue to try to read the next message
+				p.logger.Err(err).Msg("Huobi provider could not read message")
+				if err := p.ping(); err != nil {
+					p.logger.Err(err).Msg("Error sending ping")
+					if err := p.reconnect(); err != nil {
+						p.logger.Err(err).Msg("huobi provider error reconnecting")
```
Comment (adamewozniak) [inline price-feeder/oracle/provider/huobi.go:120]:
```suggestion
						p.logger.Err(err).Msg("error reconnecting to the huobi provider")
```

File: price-feeder/oracle/provider/huobi.go [hunk 5]
```diff
@@ -27,140 +32,248 @@ type (
...
-		return TickerPrice{}, fmt.Errorf("failed to make Huobi request: %w", err)
+			if len(bz) == 0 {
+				continue
+			}
+
+			p.messageReceived(messageType, bz)
+
+		case <-p.reconnectTicker.C:
+			if err := p.reconnect(); err != nil {
+				p.logger.Err(err).Msg("huobi provider error reconnecting")
```
Comment (adamewozniak) [inline price-feeder/oracle/provider/huobi.go:134]:
```suggestion
				p.logger.Err(err).Msg("error reconnecting to the huobi provider")
```

File: price-feeder/oracle/provider/huobi.go [hunk 6]
```diff
@@ -27,140 +32,248 @@ type (
...
+// reconnect closes the last WS connection and create a new one
+func (p *HuobiProvider) reconnect() error {
+	p.wsClient.Close()
 
-	bz, err := ioutil.ReadAll(resp.Body)
+	p.logger.Debug().Msg("huobi reconnecting websocket")
+	wsConn, _, err := websocket.DefaultDialer.Dial(p.wsURL.String(), nil)
 	if err != nil {
-		return sdk.ZeroDec(), fmt.Errorf("failed to read Huobi response body: %w", err)
+		return fmt.Errorf("error reconnect to huobi websocket: %w", err)
```
Comment (adamewozniak) [inline price-feeder/oracle/provider/huobi.go:217]:
```suggestion
		return fmt.Errorf("error reconnecting to huobi websocket: %w", err)
```

**Review by adamewozniak — APPROVED** [review]
just some nits, but LGTM 🚀 might want a review from @alexanderbez before merging

**Comment by RafilxTenfen** [comment]
> > @adamewozniak
> > I did 2 things to try to reconnect, one of them is as usual, try to reconnect at each 2 min if the ticker is not reset by a ping message, the other one became a little [weird](https://github.com/umee-network/umee/pull/569/files#diff-88966ac3f8eb0c4b434ceef0aac6226c51a21863af1602e40cc085ddaf14e0ecR114)
> > 
> > * if a message can not be read it generates an error in this case we try to ping the provider to check the connection
> >   
> >   * if the ping returns an error it means that the connection is closed so I try to reconnect there too
> > 
> > Any thoughts around that?
> 
> I think it makes sense to handle the case this way - I'm assuming the err coming out of `p.wsClient.ReadMessage()` is rare / would only happen if the provider was down, yes?

Usually, yes 
I had it running for some time and only generated this error when the connection was closed

File: price-feeder/oracle/provider/binance.go [hunk 7]
```diff
@@ -87,6 +87,7 @@ func NewBinanceProvider(ctx context.Context, logger zerolog.Logger, pairs ...typ
 	return provider, nil
 }
 
+// GetTickerPrices returns the tickerPrices based on the saved map
```
Comment (alexanderbez) [inline price-feeder/oracle/provider/binance.go:90]:
```suggestion
// GetTickerPrices returns the tickerPrices based on the provided pairs.
```

Comment (alexanderbez) [reply price-feeder/oracle/provider/binance.go:90]: good catch :P

File: price-feeder/oracle/provider/huobi.go [hunk 8]
```diff
@@ -27,140 +32,248 @@ type (
 	//
 	// REF: https://huobiapi.github.io/docs/spot/v1/en/#market-data
```
Comment (alexanderbez) [inline price-feeder/oracle/provider/huobi.go:33]:
```suggestion
	// REF: https://huobiapi.github.io/docs/spot/v1/en/#market-ticker
```

File: price-feeder/oracle/provider/huobi.go [hunk 9]
```diff
@@ -27,140 +32,248 @@ type (
...
-		Tick   struct {
-			Vol float64 `json:"vol"`
-		} `json:"tick"`
+		wsURL           url.URL
+		wsClient        *websocket.Conn
+		logger          zerolog.Logger
+		mu              sync.Mutex
+		tickers         map[string]HuobiTicker // market.$symbol.ticker => HuobiTicker
+		subscribedPairs []types.CurrencyPair
+		reconnectTicker *time.Ticker
```
Comment (alexanderbez) [inline price-feeder/oracle/provider/huobi.go:41]: same as the previous PR -- this doesn't really have a need to be a part of the `HuobiProvider` type.

File: price-feeder/oracle/provider/huobi.go [hunk 10]
```diff
@@ -27,140 +32,248 @@ type (
...
+	}
+
+	// HuobiTicker defines the response type for the channel and
+	// the tick object for a given ticker/symbol
```
Comment (alexanderbez) [inline price-feeder/oracle/provider/huobi.go:45]:
```suggestion
	// the tick object for a given ticker/symbol.
```

File: price-feeder/oracle/provider/huobi.go [hunk 11]
```diff
@@ -27,140 +32,248 @@ type (
...
+	HuobiTicker struct {
+		CH   string    `json:"ch"` // Data belonged channel，Format：market.$symbol.ticker
+		Tick HuobiTick `json:"tick"`
+	}
+
+	// HuobiTick defines the response type for the last 24h market summary
+	// and the last traded price for a given ticker/symbol
```
Comment (alexanderbez) [inline price-feeder/oracle/provider/huobi.go:52]:
```suggestion
	// and the last traded price for a given ticker/symbol.
```

File: price-feeder/oracle/provider/huobi.go [hunk 12]
```diff
@@ -27,140 +32,248 @@ type (
...
+		wsClient:        wsConn,
+		logger:          logger.With().Str("module", "oracle").Logger(),
+		tickers:         map[string]HuobiTicker{},
+		reconnectTicker: time.NewTicker(huobiReconnectTime),
+	}
+
+	if err := provider.subscribeTickers(pairs...); err != nil {
+		return nil, err
+	}
+	provider.subscribedPairs = pairs
```
Comment (alexanderbez) [inline price-feeder/oracle/provider/huobi.go:87]: why not have this be part of the struct literal creation on line 76?

Comment (alexanderbez) [reply price-feeder/oracle/provider/huobi.go:87]: You can update the other providers too :)

File: price-feeder/oracle/provider/huobi.go [hunk 13]
```diff
@@ -27,140 +32,248 @@ type (
...
 }
 
-func (p HuobiProvider) getTickerPrice(ticker string) (TickerPrice, error) {
-	path := fmt.Sprintf("%s%s?symbol=%s", p.baseURL, huobiTradeEndpoint, strings.ToLower(ticker))
+func (p *HuobiProvider) handleWebSocketMsgs(ctx context.Context) {
+	for {
+		select {
+		case <-ctx.Done():
+			return
+		case <-time.After(defaultReadNewMessage):
```
Comment (alexanderbez) [inline price-feeder/oracle/provider/huobi.go:111]: Can we rename this to `defaultReadNewWSMessage`?

File: price-feeder/oracle/provider/huobi.go [hunk 14]
```diff
@@ -27,140 +32,248 @@ type (
...
+			// time after to avoid asking for prices too frequently
```
Comment (alexanderbez) [inline price-feeder/oracle/provider/huobi.go:112]: What does this comment mean?

File: price-feeder/oracle/provider/huobi.go [hunk 15]
```diff
@@ -27,140 +32,248 @@ type (
...
+			messageType, bz, err := p.wsClient.ReadMessage()
+			if err != nil {
+				// if some error occurs continue to try to read the next message
+				p.logger.Err(err).Msg("Huobi provider could not read message")
```
Comment (alexanderbez) [inline price-feeder/oracle/provider/huobi.go:116]:
```suggestion
				p.logger.Err(err).Msg(" failed to read message from Huobi provider")
```

File: price-feeder/oracle/provider/huobi.go [hunk 16]
```diff
@@ -27,140 +32,248 @@ type (
...
+				if err := p.ping(); err != nil {
+					p.logger.Err(err).Msg("Error sending ping")
```
Comment (alexanderbez) [inline price-feeder/oracle/provider/huobi.go:118]:
```suggestion
					p.logger.Err(err).Msg("failed to send ping")
```

File: price-feeder/oracle/provider/huobi.go [hunk 17]
```diff
@@ -27,140 +32,248 @@ type (
...
+					if err := p.reconnect(); err != nil {
+						p.logger.Err(err).Msg("error reconnecting to the huobi provider")
```
Comment (alexanderbez) [inline price-feeder/oracle/provider/huobi.go:120]:
```suggestion
						p.logger.Err(err).Msg("error reconnecting to the Huobi provider")
```

File: price-feeder/oracle/provider/huobi.go [hunk 18]
```diff
@@ -27,140 +32,248 @@ type (
...
-		return TickerPrice{}, fmt.Errorf("failed to make Huobi request: %w", err)
+			if len(bz) == 0 {
+				continue
+			}
+
+			p.messageReceived(messageType, bz)
+
+		case <-p.reconnectTicker.C:
+			if err := p.reconnect(); err != nil {
+				p.logger.Err(err).Msg("error reconnecting to the huobi provider")
```
Comment (alexanderbez) [inline price-feeder/oracle/provider/huobi.go:134]:
```suggestion
				p.logger.Err(err).Msg("error reconnecting to the Huobi provider")
```

File: price-feeder/oracle/provider/huobi.go [hunk 19]
```diff
@@ -27,140 +32,248 @@ type (
...
+			}
+		}
 	}
+}
 
-	defer resp.Body.Close()
+// messageReceived handles the received data from Huobi
```
Comment (alexanderbez) [inline price-feeder/oracle/provider/huobi.go:140]:
```suggestion
// messageReceived handles the received data from the Huobi websocket.
```

File: price-feeder/oracle/provider/huobi.go [hunk 20]
```diff
@@ -27,140 +32,248 @@ type (
...
+func (p *HuobiProvider) messageReceived(messageType int, bz []byte) {
+	if messageType != websocket.BinaryMessage {
+		return
+	}
 
-	bz, err := ioutil.ReadAll(resp.Body)
+	bz, err := uncompressGzip(bz)
 	if err != nil {
-		return TickerPrice{}, fmt.Errorf("failed to read Huobi response body: %w", err)
+		p.logger.Err(err).Msg("Huobi provider could not uncompress gzip message")
```
Comment (alexanderbez) [inline price-feeder/oracle/provider/huobi.go:150]:
```suggestion
		p.logger.Err(err).Msg("failed to decompress Huobi gziped message")
```

File: price-feeder/oracle/provider/huobi.go [hunk 21]
```diff
@@ -27,140 +32,248 @@ type (
...
+				p.logger.Err(err).Msg("error reconnecting to the huobi provider")
+			}
+		}
 	}
+}
 
-	defer resp.Body.Close()
+// messageReceived handles the received data from Huobi
+// All return data of websocket Market APIs are compressed with
+// GZIP so they need to be unzipped.
```
Comment (alexanderbez) [inline price-feeder/oracle/provider/huobi.go:142]:
```suggestion
// GZIP so they need to be decompressed.
```

File: price-feeder/oracle/provider/huobi.go [hunk 22]
```diff
@@ -27,140 +32,248 @@ type (
...
 	if err != nil {
-		return TickerPrice{}, fmt.Errorf("failed to read Huobi response body: %w", err)
+		p.logger.Err(err).Msg("Huobi provider could not uncompress gzip message")
+		return
 	}
 
-	var tradeResp HuobiTraceResponse
-	if err := json.Unmarshal(bz, &tradeResp); err != nil {
-		return TickerPrice{}, fmt.Errorf("failed to unmarshal Huobi response body: %w", err)
+	if bytes.Contains(bz, []byte("ping")) {
```
Comment (alexanderbez) [inline price-feeder/oracle/provider/huobi.go:154]: let's put `[]byte("ping")` in a global private `var`?

File: price-feeder/oracle/provider/huobi.go [hunk 23]
```diff
@@ -27,140 +32,248 @@ type (
...
 
-	if !strings.EqualFold(tradeResp.Status, "ok") {
-		return TickerPrice{}, fmt.Errorf(
-			"received unexpected error from Huobi response: %s",
-			tradeResp.ErrMsg,
-		)
+	var tickerResp HuobiTicker
+	if err := json.Unmarshal(bz, &tickerResp); err != nil {
+		// sometimes it returns other messages which are not ticker responses
+		p.logger.Err(err).Msg("Huobi provider could not unmarshal")
```
Comment (alexanderbez) [inline price-feeder/oracle/provider/huobi.go:162]:
```suggestion
		p.logger.Err(err).Msg("failed to unmarshal message from Huobi provider")
```

File: price-feeder/oracle/provider/huobi.go [hunk 24]
```diff
@@ -27,140 +32,248 @@ type (
...
 
-	rawVolume := marketResp.Tick.Vol
-	volume, err := sdk.NewDecFromStr(strconv.FormatFloat(rawVolume, 'f', -1, 64))
+	return ticker.toTickerPrice()
+}
+
+// uncompressGzip uncompress gzip compressed messages
+// All data returned from the websocket Market APIs is compressed
+// with GZIP, so it needs to be unzipped.
+func uncompressGzip(bz []byte) ([]byte, error) {
```
Comment (alexanderbez) [inline price-feeder/oracle/provider/huobi.go:251]:
```suggestion
func decompressGzip(bz []byte) ([]byte, error) {
```

File: price-feeder/oracle/provider/huobi.go [hunk 25]
```diff
@@ -27,140 +32,248 @@ type (
...
+	if !ok {
+		return TickerPrice{}, fmt.Errorf("failed to get ticker price for %s", cp.String())
 	}
 
-	rawVolume := marketResp.Tick.Vol
-	volume, err := sdk.NewDecFromStr(strconv.FormatFloat(rawVolume, 'f', -1, 64))
+	return ticker.toTickerPrice()
+}
+
+// uncompressGzip uncompress gzip compressed messages
```
Comment (alexanderbez) [inline price-feeder/oracle/provider/huobi.go:248]:
```suggestion
// decompressGzip uncompress gzip compressed messages
```

**Review by alexanderbez — CHANGES_REQUESTED** [review]
Looks great @RafilxTenfen. Just left a bit of minor nitpick feedback ;-)

File: price-feeder/oracle/provider/huobi.go (hunk 9 above)
Comment (RafilxTenfen) [reply price-feeder/oracle/provider/huobi.go]: done at https://github.com/umee-network/umee/pull/569/commits/576f19d2c689b558508636c8419600b5c3c75aaf

File: price-feeder/oracle/provider/huobi.go (hunk 12 above)
Comment (RafilxTenfen) [reply price-feeder/oracle/provider/huobi.go]: done at https://github.com/umee-network/umee/pull/569/commits/9c7f0c6af5cf96bbafdb3373fe52aca44e51b2b7

File: price-feeder/oracle/provider/huobi.go (hunk 13 above)
Comment (RafilxTenfen) [reply price-feeder/oracle/provider/huobi.go]: done at https://github.com/umee-network/umee/pull/569/commits/d90973ee0251cf9d4e4caf1b6384192d10084a36

File: price-feeder/oracle/provider/huobi.go (hunk 14 above)
Comment (RafilxTenfen) [reply price-feeder/oracle/provider/huobi.go]: It was kind of to give a reason to why have the `time.After` 
Removed at https://github.com/umee-network/umee/pull/569/commits/9269b0c910897b95d3ce3a279368d7e45b31339a

File: price-feeder/oracle/provider/huobi.go (hunk 22 above)
Comment (RafilxTenfen) [reply price-feeder/oracle/provider/huobi.go]: done at https://github.com/umee-network/umee/pull/569/commits/e9099221b3884748aed6f8862b70836b3a2270e7

File: price-feeder/oracle/provider/huobi.go [hunk 26]
```diff
@@ -25,142 +30,249 @@ type (
...
+	for {
+		select {
+		case <-ctx.Done():
+			return
+		case <-time.After(defaultReadNewWSMessage):
+			messageType, bz, err := p.wsClient.ReadMessage()
+			if err != nil {
+				// if some error occurs, check if connection is alive
+				// and continue to try to read the next message
+				p.logger.Err(err).Msg(" failed to read message from Huobi provider")
```
Comment (alexanderbez) [inline price-feeder/oracle/provider/huobi.go:115]:
```suggestion
				p.logger.Err(err).Msg("failed to read message from Huobi provider")
```

File: price-feeder/oracle/provider/huobi.go [hunk 27]
```diff
@@ -25,142 +30,249 @@ type (
...
+	reconnectTicker := time.NewTicker(huobiReconnectTime)
+	for {
+		select {
+		case <-ctx.Done():
+			return
+		case <-time.After(defaultReadNewWSMessage):
+			messageType, bz, err := p.wsClient.ReadMessage()
+			if err != nil {
+				// if some error occurs, check if connection is alive
+				// and continue to try to read the next message
```
Comment (alexanderbez) [inline price-feeder/oracle/provider/huobi.go:114]:
```suggestion
				// If some error occurs, check if connection is alive
				// and continue to try to read the next message.
```

**Review by alexanderbez — APPROVED** [review]
Nice work @RafilxTenfen 👏

**Comment by alexanderbez** [comment]
> @Mergifyio backport release/price-feeder/v0.1.x

@RafilxTenfen you only need to add this comment _iff_ the PR was already _merged_. While it's still open, you can just simply add a `backport` label to the PR (I added it). 👍

**Comment by RafilxTenfen** [comment]
> > @Mergifyio backport release/price-feeder/v0.1.x
> 
> @RafilxTenfen you only need to add this comment _iff_ the PR was already _merged_. While it's still open, you can just simply add a `backport` label to the PR (I added it). +1

Thanks!

**Pull request merged**